### PR TITLE
Add verify-deps for go mod

### PR DESCRIPTION
**Pull request by tnozicka**
The current verify-deps was built for glide. This commit is adding
automatic modules detection and if detected it uses the new
implementation based on go mod.

All that's left is to revert the verify deps job removal ;)

/cc @sttts @soltysh @mfojtik 
fyi @openshift/sig-master

### CODE REVIEW

File: alpha-build-machinery/make/targets/openshift/deps-gomod.mk [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+# We need to force localle so different envs sort files the same way for recursive traversals
+deps_diff :=LC_COLLATE=C diff --no-dereference -N
+
+# $1 - temporary directory
+define restore-deps
+	ln -s $(abspath ./) "$(1)"/current
+	cp -R -H ./ "$(1)"/updated
+	$(RM) -r "$(1)"/updated/vendor
+	cd "$(1)"/updated && $(GO) mod tidy && $(GO) mod vendor && $(GO) mod verify
```
Comment (soltysh) [inline alpha-build-machinery/make/targets/openshift/deps-gomod.mk]: Shouldn't the order by vendor and then tidy?

Comment (tnozicka) [reply alpha-build-machinery/make/targets/openshift/deps-gomod.mk]: As it modifies the go.mod I see `go mod tidy` as first step. It also adds dependencies and prune them which should logically preceed commit them to vendor folder.
```
go mod tidy finds all the packages transitively imported by packages in your module. It adds new module requirements for packages not provided by any known module, and it removes requirements on modules that don't provide any imported packages. If a module provides packages that are only imported by projects that haven't migrated to modules yet, the module requirement will be marked with an // indirect comment. It is always good practice to run go mod tidy before committing a go.mod file to version control.
```
https://blog.golang.org/migrating-to-go-modules

**Comment by deads2k** [comment]
/approve
/hold

hold for proof.  Please link proof PR.

**Comment by tnozicka** [comment]
@deads2k That's fair. Making a proof and also gating this PR in https://github.com/openshift/release/pull/6458

**Comment by tnozicka** [comment]
Yeah, I can't open a proof PR because the verify-deps CI job was removed from those repos. I can't bring it back for this repo without blocking all PRa on this one. So this needs to logically go first.

I've run the verify job from https://github.com/openshift/release/pull/6458 manually against this branch which should provide the confidence you asked for:
```
$ go run ../ci-tools/cmd/ci-operator/main.go --config ./ci-operator/config/openshift/library-go/openshift-library-go-master.yaml --git-ref=tnozicka/library-go@gomod-verify --target=verify-deps
2019/12/18 10:14:02 Resolved tnozicka/library-go@gomod-verify to commit a97f51c5ff848ef4ff9c6cf81f94dd5362132e57
2019/12/18 10:14:02 Resolved source https://github.com/tnozicka/library-go to gomod-verify@a97f51c5
2019/12/18 10:14:03 Resolved openshift/release:golang-1.13 to sha256:8d486ce12512b80b337630d35b0807c7ba863567759e4b75966479621c141fba
2019/12/18 10:14:03 Resolved openshift/centos:7 to sha256:285bc3161133ec01d8ca8680cd746eecbfdbc1faa6313bd863151c4b26d7e5a5
2019/12/18 10:14:03 Using namespace ci-op-18ripv4h
2019/12/18 10:14:03 Running [input:root], src, bin, verify-deps
2019/12/18 10:14:03 Creating namespace ci-op-18ripv4h
2019/12/18 10:14:03 Setting a soft TTL of 1h0m0s for the namespace
2019/12/18 10:14:03 Setting a hard TTL of 12h0m0s for the namespace
2019/12/18 10:14:03 warning: Could not add annotations because you do not have permission to update the namespace (details: namespaces "ci-op-18ripv4h" is forbidden: User "tnozicka" cannot update namespaces in the namespace "ci-op-18ripv4h": no RBAC policy matched)
2019/12/18 10:14:03 Setting up pipeline imagestream for the test
2019/12/18 10:14:04 warning: Events will not be created because of lack of permission
2019/12/18 10:14:04 Tagging openshift/release:golang-1.13 into pipeline:root
2019/12/18 10:14:04 Building src
2019/12/18 10:14:38 Build src succeeded after 34s
2019/12/18 10:14:38 Building bin
2019/12/18 10:15:56 Build bin succeeded after 1m18s
2019/12/18 10:15:56 Executing test verify-deps
2019/12/18 10:16:36 Container test in pod verify-deps completed successfully
2019/12/18 10:16:36 Pod verify-deps succeeded after 38s
2019/12/18 10:16:37 Ran for 2m34s
```
/hold cancel

**Pull request merged**